password-hash/src/salt.rs

```diff
@@ -134,6 +135,17 @@ impl<'a> Salt<'a> {
     }
 }
 
+fn map_error_to_salt_error(err: Error) -> Error {
+    match err {
+        Error::ParamValueInvalid(value_err) => match value_err {
+            ParamValueError::ToLong => Error::SaltTooLong,
+            ParamValueError::ToShort | ParamValueError::NotProvided => Error::SaltTooShort,
+            _ => Error::SaltInvalid,
+        },
+        _ => Error::SaltInvalid,
+    }
+}
```

Comment:
Seems like you could get rid of this if the two error types shared a common inner error enum type as suggested above.

Reply:
Not sure if we can get rid of it completely but it definitely simplifies a bit. I have removed the new method and did it inline as it is simpler now. Please let me know in case I have missed some way to do it better 🤔